Automattic/Gravatar-SDK-Android
# QuickEditor: Delete avatar

Closes #236

### Description

This PR implements the base of the delete avatar flow. 

**Note:** When opening this PR, the endpoints are not working yet, so you'll always receive a `404`; however, I think testing the whole flow is possible. However, it's possible that we'll need to make minor adjustments when the endpoint is released.

https://github.com/user-attachments/assets/7cc13327-c440-469f-8d13-61aa6ffb3d42

This PR will be followed by two more to add the confirmation dialog (#449)  and the error toast (#448).

### Testing Steps

1. Open the QE
2. In any of your uploaded avatars, tap the options button (...)
3. Select `Delete`
4. You should see how the avatar disappears immediately (as we are assuming the flow will work)
5. You should see the avatar reappear when we receive the `404` (that's the work part of not having the endpoints yet :) )
6. If you want, you can check logcat a see the error:
```
2024-11-20 12:05:40.816 10320-10465 AvatarService           com.gravatar.demoapp                 W  Network call unsuccessful trying to delete Gravatar avatar: Response{protocol=h2, code=404, message=, url=https://api.gravatar.com/v3/me/avatars/bfb35130743eb019b309a3c219ee77b5}.body
```
7. You can also play with a network interceptor and mock a 200 answer (I don't think it's completely necessary)
